### DISCUSSION TITLE
systemd unit errors in nginx proxy manager

### DISCUSSION THREAD

**Discussion by githubsean** [tteck/Proxmox#2986] [General]
TLDR: It appears some systemd units are being started (and they fail) when they should not be started within a Linux container.

I connected to the NPM LXC
`pct exec 101 /bin/bash`

I then ran `systemctl status`, and notice there were some errors. The issue was with two units:
- systemd-modules-load.service
- systemd-journald-audit.socket

Out of interest I created an Ubuntu 22.04 LXC using the built in template and noticed there were no startup issues - i.e. `systemctl status` showed "running". I then pulled up the status for these units in the Ubuntu LXC and it showed they were not started because a condition was not met.
I then had a look at the systemd unit files for these two units in /lib/systemd/system and compared the unit files between the ubuntu LXC and the npm LXC. Each of these unit files had an additional line:
- systemd-modules-load.service: `ConditionVirtualization=!container`
- systemd-journald-audit.socket: `ConditionVirtualization=!private-users`

I added these conditions to these modules, this meant they didn't start, and therefore there is a clear run status. 

I don't know which base of (Debian?) was used for NPM - maybe it does not have these extra virtualisation aware lines?
Rather than editing/adding these lines in the base OS config, it would also be possible to add overriding snippets to a directory named the same as the unit file to /etc/systemd/system.

Other than the errors that show in the running status, there I don't think there is any actual issue with these units failing to start, but I like to have a clean system so when/if something does throw an error, it is something to investigate.

**Comment by tteck** [comment]
Why are you connecting to the container this way?

**Reply by githubsean** [reply]
Forgot the password
